content/client-lib-development-guide/features.textile

```diff
@@ -1697,13 +1735,40 @@ class RealtimeChannel:
   unsubscribe(String, (Message) ->) // RTL8a
   setOptions(options: ChannelOptions) => io // RTL16
 
+class BatchOperations:
+  publish([BatchSpec]) => BatchResult<BatchPublishResponse> // BO2a
+  publish(BatchSpec) => BatchResult<BatchPublishResponse> // BO2a
+  getPresence([String]) => BatchResult<BatchPresenceResponse> // BO2b
+
+class BatchResult<T>:
+  error: ErrorInfo? // BPA2b
+  responses: []T? // BPA2a
+
+class BatchPublishResponse:
+  channel: String // BPB2a
+  messageId: String? // BPB2b
+  error: ErrorInfo? // BPB2c
+
+class BatchPresenceResponse:
+  channel: String // BPD2a
+  presence: []BatchPresence // PBD2b
+
+class BatchPresence:
+  clientId: string
+  action: string?
+  error: ErrorInfo?
+
 class PushChannel:
   subscribeDevice() => io // RSH7a
   subscribeClient() => io // RSH7b
   unsubscribeDevice() => io // RSH7c
   unsubscribeClient() => io // RSH7d
   listSubscriptions() => io PaginatedResult<PushChannelSubscription> // RSH7e
 
+class BatchSpec:
```

Comment:
I know the documentation calls this `BatchSpec` too, but it seems specific to publishing a batch of messages, so perhaps it should be called something like `BatchPublishRequest`?

Reply:
The [batch publish documentation](https://ably.com/docs/rest/batch#publish-requests) reads as if the BatchSpec object is the one passed into the body of the request either as a single object or within an array as it's being used here. In any case I would argue that the spec should be naming things as consistently as possible with the REST documentation.

Reply:
> In any case I would argue that the spec should be naming things as consistently as possible with the REST documentation.

I agree with that, so I guess what I'm saying is, how about we rename it to `BatchPublishRequest` in both places? This term isn't something that appears in the API response itself, so we should be free to change it, and now would be the time to change it if we think it could be named better before it starts to appear in SDK code.

Reply:
I've renamed it to `BatchPublishRequest` both in the spec and in the docs in [a2eee7d](https://github.com/ably/docs/pull/1450/commits/a2eee7de39d8b63bf0a706919e75f90dfdfd9381)